src/input/process/parameter.rs

```diff
@@ -112,48 +114,70 @@ fn read_process_parameters_from_iter<I>(
 where
     I: Iterator<Item = ProcessParameterRaw>,
 {
-    let mut params: HashMap<ProcessID, ProcessParameterMap> = HashMap::new();
+    let mut map: HashMap<ProcessID, ProcessParameterMap> = HashMap::new();
     for param_raw in iter {
+        // Get process
         let id = process_ids.get_id_by_str(&param_raw.process_id)?;
-
-        let entry = params.entry(id.clone()).or_default();
         let process = processes
             .get(&id)
             .with_context(|| format!("Process {id} not found"))?;
+
+        // Get years
         let process_year_range = &process.years;
         let process_years: Vec<u32> = milestone_years
             .iter()
             .copied()
             .filter(|year| process_year_range.contains(year))
             .collect();
-
         let parameter_years =
             parse_year_str(&param_raw.year, &process_years).with_context(|| {
                 format!("Invalid year for process {id}. Valid years are {process_years:?}")
             })?;
+
+        // Get regions
+        let process_regions = process.regions.clone();
+        let parameter_regions = parse_region_str(&param_raw.regions, &process_regions)
+            .with_context(|| {
+                format!("Invalid region for process {id}. Valid regions are {process_regions:?}")
+            })?;
+
+        // Insert parameter into the map
         let param = param_raw.into_parameter()?;
+        let entry = map.entry(id.clone()).or_default();
         for year in parameter_years {
-            try_insert(entry, year, param.clone())?;
+            for region in parameter_regions.clone() {
+                try_insert(entry, (region, year), param.clone())?;
+            }
         }
     }
 
-    // Check parameters cover all years of the process
-    for (id, parameter) in params.iter() {
-        let year_range = &processes.get(id).unwrap().years;
+    // Check parameters cover all years and regions of the process
+    for (id, parameters) in map.iter() {
+        let process = processes.get(id).unwrap();
+        let year_range = process.years.clone();
         let reference_years: HashSet<u32> = milestone_years
             .iter()
             .copied()
             .filter(|year| year_range.contains(year))
             .collect();
-        let parameter_years: HashSet<u32> = parameter.keys().copied().collect();
+        let reference_regions = process.regions.clone();
+
+        let mut missing_keys = Vec::new();
+        for year in &reference_years {
+            for region in &reference_regions {
+                if !parameters.contains_key(&(region.clone(), *year)) {
+                    missing_keys.push((region, *year));
+                }
```

Comment:
I think this should work:

```suggestion
                let key = (region, *year);
                if !parameters.contains_key(&key) {
                    missing_keys.push(key);
                }
```